posts/2024-04-15-simplifying-nosql-database-integration-with-jakarta-nosql.adoc

```diff
@@ -0,0 +1,418 @@
+[.img_border_light]
+image::/img/blog/document-nosql.png[Document database type,align="center" width=25%,height=25%]
+
+Key vendors providing document database solutions are: *MongoDB*, *Couchbase*, *Elastic*, *Oracle NoSQL Database*.
+
+===== Column Databases
+
+The Column databases (also known as column-oriented or wide-column databases) store data as columns instead of rows (common in traditional relational databases). This approach is a differentiator across other types, as it's an efficient way to handle large amounts of data and run performant complex queries.
+
+This type is designed and optimized for storing large amounts of structured, semi-structured, and unstructured data with a flexible schema, and supports high levels of concurrency and scalability.
+
+Wide-column databases are often used for analytics, content management, and data warehousing.
+
+[.img_border_light]
+image::/img/blog/column-nosql.png[Column database type,align="center" width=25%,height=25%]
+
+Examples of Column databases on the market include *Apache HBase*, *Apache Cassandra*, *Scylla*, *Azure Cosmos DB*, and many others. The first two mentioned here are open source technologies.
+
+===== Graph Databases
+
+The Graph NoSQL database is optimized for storing and querying data with complex relationships. In this approach, data is managed as a graph where entities can be represented as nodes and edges, resulting in performant management of complex relationship and hierarchies.
+
+The graph data resembles the graph of objects in the Oriented-Object Programming (OOP) paradigm. Graph NoSQL database solutions is a good fit for scenarios that require fast querying of highly interconnected data, such as social networks, recommendation engines, and fraud detection systems.
+
+[.img_border_light]
+image::/img/blog/graph-nosql.png[Graph database type,align="center" width=25%,height=25%]
+
+Engines to choose from today, are *Neo4J*, *Arango DB*, *OrientDB*, *JanusGraph*, among others. The last one mentioned is open source technologies.
+
+==== Challenges of NoSQL Integration
+
+Modern solution's requirements frequently requires the capabilities and benefits of different types of NoSQL databases, therefore, we should be able to work with multiple NoSQL solutions, coming from different vendors. Having said that, we can expect to face challenges such as:
+
+* A high cognitive load necessary when choosing a NoSQL database for their solutions;
+* A significant learning curve derived from usage of different database APIs;
+* Extra time invested on changing/maintaining existing codebase;
+* Potentially increase in complexity when onboarding new developers to the team;
+
+Furthermore, in the cloud era and the consumption-based pricing models, we must consider ways to deliver solutions with efficient resource consumption for costs reduction. This is how potential discussions may rise, opening possibility for considering changes of the underlying database used by existing applications.
```

Comment:
Consider changing from "This is how potential discussions may rise," to "This is how potential discussions may arise,"